src/Microsoft.OData.Core/Microsoft.OData.Core.txt

```diff
@@ -120,7 +120,7 @@ ODataParameterWriterCore_CannotWriteParameter=WriteValue and CreateCollectionWri
 ODataParameterWriterCore_CannotWriteEnd=WriteEnd can only be called after WriteStart and after the previously created sub-writer has completed.
 ODataParameterWriterCore_CannotWriteInErrorOrCompletedState=The writer is in either the 'Error' or 'Completed' state. No further writes can be performed on this writer.
 ODataParameterWriterCore_DuplicatedParameterNameNotAllowed=The parameter '{0}' has already been written. Duplicate parameter names are not allowed in the parameter payload.
-ODataParameterWriterCore_CannotWriteValueOnNonValueTypeKind=The parameter '{0}' is of Edm type kind '{1}'. You cannot call WriteValue on a parameter that is not of Edm type kinds 'Primitive', 'Enum' or 'Complex'.
+ODataParameterWriterCore_CannotWriteValueOnNonValueTypeKind=The parameter '{0}' is of Edm type kind '{1}'. You cannot call WriteValue on a parameter that is not of Edm type kinds 'Primitive' or 'Enum'.

```

Comment:
>ODataParameterWriterCore_Canno [](start = 0, length = 30)

if you change the "txt' file, please right click the 
Microsoft.OData.Core.tt 
and
Parameterized.Microsoft.OData.Core.tt
in the Visual studio, select "run the custom tool" to re-generate the corresponding CS file. otherwise, the error message changes can't refresh into the assembly.

Meanwhile, is there any Unit test cases in "OData.Core.Test" project to verify the error message. If no, please add one to verify that.

Reply:
@xuzhg We have not added or removed any string variable. We have only changed the value therefore running the `.tt` files do not change the content of the corresponding `.cs` files.


[Yes, we have tests to verify exception being thrown.](https://github.com/OData/odata.net/blob/a0fc496c360d4feda0f2a27cb1494c52a07e7fe4/test/FunctionalTests/Tests/DataOData/Tests/OData.Writer.Tests/ParameterWriter/ParameterWriterTests.cs#L874-L1207)